orttraining/orttraining/training_api/checkpoint.cc

```diff
@@ -1,164 +1,94 @@
-    const OrtValue& ort_value = name_to_ort_value.at(tensor_name);
-    ORT_RETURN_IF_NOT(ort_value.IsTensor(), "ort_value.IsTensor() was false");
-    const Tensor& src_tensor = ort_value.Get<Tensor>();
-
-    // Currently large model size not considered, so exception thrown here
-    // when protobuf upper limit hit.
-    total_bytes += static_cast<uint64_t>(src_tensor.SizeInBytes());
-    if (total_bytes >= PROTOBUF_UPPER_LIMIT) {
-      ORT_THROW("checkpoint file size hit upper limit.");
-    }
-
-    saved_tensor_protos.emplace_back(utils::CopyTensorToTensorProto(
-        src_tensor, tensor_name, data_transfer_manager));
+    flatbuffers::FlatBufferBuilder& builder,
+    std::vector<flatbuffers::Offset<fbs::Tensor>>& flatbuffer_tensors) {
+  for (const auto& [name, ort_value] : name_to_ort_value) {
+    flatbuffers::Offset<fbs::Tensor> fbs_tensor;
+    ORT_RETURN_IF_ERROR(fbs::utils::SaveOrtValueOrtFormat(name, ort_value, data_transfer_manager, builder, fbs_tensor));
+    flatbuffer_tensors.emplace_back(fbs_tensor);
   }
 
   return Status::OK();
 }
 
-PathString GetTensorProtoFilePath(const PathString& checkpoint_directory, const PathString& filename_prefix) {
-  std::basic_ostringstream<PathChar> oss;
-  oss << filename_prefix << k_name_separator << k_tensor_proto_file_name;
-  return ConcatPathComponent<PathChar>(checkpoint_directory, oss.str());
-}
-
-PathString GetTensorProtoPropertiesFilePath(
-    const PathString& checkpoint_directory, const PathString& filename_prefix) {
-  std::basic_ostringstream<PathChar> oss;
-  oss << filename_prefix << k_name_separator << k_tensor_proto_properties_file_name;
-  return ConcatPathComponent<PathChar>(checkpoint_directory, oss.str());
-}
-
-PathString StringConcat(
-    const PathString& s_a, const PathString& s_b,
-    const PathString& del = k_name_separator) {
-  std::basic_ostringstream<PathChar> oss;
-  oss << s_a << del << s_b;
-  return oss.str();
-}
-
-void StringSplit(const PathString& s, std::vector<PathString>& results,
-                 const PathString& del = k_name_separator) {
-  ORT_ENFORCE(!s.empty(), "String to split is empty");
-  size_t start = 0;
-  size_t end = s.find(del);
-  while (end != std::string::npos) {
-    results.push_back(s.substr(start, end - start));
-    start = end + del.size();
-    end = s.find(del, start);
+/**
+ * @brief Create OrtValue objects from flatbuffer tensors.
+ *
+ * @param flatbuffer_tensors Flatbuffer tensors.
+ * @param name_to_ort_value Name to OrtValue map to be populated.
+ * @return Status of the operation.
+ */
+Status OrtValuesFromFlatbufferTensors(
+    const flatbuffers::Vector<flatbuffers::Offset<onnxruntime::fbs::Tensor>>* flatbuffer_tensors,
+    InlinedHashMap<std::string, OrtValue>& name_to_ort_value) {
+  ORT_RETURN_IF_NOT(flatbuffer_tensors, "Expected: Both trainable and non trainable tensors must exist.",
+                    " Actual: Encountered a nullptr. Checkpoint file is invalid");
```

Comment:
nit: If you require an argument to not be null make it a reference. That makes the expected usage of the function clear to the reader and you don't need the ORT_RETURN_IF_NOT.

You may need to do the check for not null at a higher level, but the error message there is most likely going to be more meaningful and easily understood than the fairly generic 'OrtValuesFromFlatbufferTensors' returning an error about 'Both trainable and non trainable tensors' which is hard to understand as there's a lack of context.